### PR TITLE
typing: update MISSING and incorrect typehints

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary
- add some missing overloads, (PartialMessage.edit, ForumChannel.create_thread, CategoryChannel.move)
- use `...` in overloads instead of `MISSING`
- use a slightly better exception on CategoryChannel.create_X_channel methods when category is passed.
- typehint topic as Optional where necessary

## Checklist
- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -1991,6 +1991,50 @@ async def edit(self, *, reason=None, **options):
...
+        ...
+
+    @overload
+    async def move(
+        self,
+        *,
+        end: bool,
+        offset: int = ...,
+        sync_permissions: bool = ...,
+        reason: str = ...,
```
Comment (4310V343k) [inline disnake/channel.py:2012]: Optional

File: disnake/channel.py [hunk 2]
```diff
@@ -1991,6 +1991,50 @@ async def edit(self, *, reason=None, **options):
...
+        ...
+
+    @overload
+    async def move(
+        self,
+        *,
+        before: Snowflake,
+        offset: int = ...,
+        sync_permissions: bool = ...,
+        reason: str = ...,
```
Comment (4310V343k) [inline disnake/channel.py:2023]: Optional

File: disnake/channel.py [hunk 3]
```diff
@@ -1991,6 +1991,50 @@ async def edit(self, *, reason=None, **options):
...
+        ...
+
+    @overload
+    async def move(
+        self,
+        *,
+        after: Snowflake,
+        offset: int = ...,
+        sync_permissions: bool = ...,
+        reason: str = ...,
```
Comment (4310V343k) [inline disnake/channel.py:2034]: Optional

Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/482/commits/a2c21e33329b9b1a3e13392407caa7958c102215

File: disnake/channel.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/482/commits/a2c21e33329b9b1a3e13392407caa7958c102215

File: disnake/abc.py [hunk 4]
```diff
@@ -959,10 +959,10 @@ async def move(
...
         *,
         after: Snowflake,
-        offset: int = MISSING,
-        category: Optional[Snowflake] = MISSING,
-        sync_permissions: bool = MISSING,
-        reason: str = MISSING,
+        offset: int = ...,
+        category: Optional[Snowflake] = ...,
+        sync_permissions: bool = ...,
+        reason: Optional[str] = ...,
```
Comment (shiftinv) [inline disnake/abc.py:965]: reason type in docstring should be optional as well

Comment (onerandomusername) [reply disnake/abc.py:965]: https://github.com/DisnakeDev/disnake/pull/482/commits/51c5f84eaceae0187de2843967e4eb74da14b201

**Review by shiftinv — APPROVED** [review]

**Comment by shiftinv** [comment]
This now has some conflicts due to #474, and a few overloads will have to be updated accordingly as well

**Comment by shiftinv** [comment]
blocked by #512

File: disnake/channel.py [hunk 5]
```diff
@@ -2418,6 +2470,78 @@ def get_thread(self, thread_id: int, /) -> Optional[Thread]:
         """
         return self.guild.get_thread(thread_id)
 
+    @overload
```
Comment (shiftinv) [inline disnake/channel.py:2473]: overloads are missing `suppress_embeds`

Comment (onerandomusername) [reply disnake/channel.py:2473]: Its probably fine

Comment (onerandomusername) [reply disnake/channel.py:2473]: https://github.com/DisnakeDev/disnake/pull/482/commits/77c393830121f1e975b1445e7476ddf433e3444f

File: disnake/message.py [hunk 6]
```diff
@@ -2061,6 +2061,70 @@ async def fetch(self) -> Message:
...
 
+    @overload
+    async def edit(
+        self,
+        content: Optional[str] = ...,
+        *,
+        embed: Optional[Embed] = ...,
+        file: File = ...,
+        attachments: Optional[List[Attachment]] = ...,
+        suppress: bool = ...,
```
Comment (shiftinv) [inline disnake/message.py:2072]:
```suggestion
        suppress_embeds: bool = ...,
```
+ other overloads

Comment (onerandomusername) [reply disnake/message.py]: https://github.com/DisnakeDev/disnake/pull/482/commits/625a69504facd183e2adb35786b3c0d72361ab86

**Pull request merged**